qiita_pet/handlers/study_handlers/listing_handlers.py

```diff
@@ -168,10 +174,16 @@ class SearchStudiesAJAX(BaseHandler):
     def get(self, ignore):
         user = self.get_argument('user')
         query = self.get_argument('query')
+        search_type = self.get_argument('search_type')
         echo = int(self.get_argument('sEcho'))
 
         if user != self.current_user.id:
             raise HTTPError(403, 'Unauthorized search!')
+        if search_type not in ['user', 'public']:
+            self.clear()
```

Comment:
In that case you should be able to use the code above - is what I've been using in the qiita_db handlers. If the error message is not sent, then there is a problem in the function `write_error` on the base handler. I think the one in the qiita_db subdir is different and does it correctly.